src/pokemon.c

```diff
@@ -8254,6 +8254,20 @@ u16 GetFormChangeTargetSpeciesBoxMon(struct BoxPokemon *mon, u16 method, u32 arg
                         }
                     }
                     break;
+                case FORM_TIME:
+                    RtcCalcLocalTime();
+                    switch (formChanges[i].param1)
+                    {
+                    case DAY:
+                        if (gLocalTime.hours >= 12 && gLocalTime.hours < 24)
```

Comment:
I think it would be better to define constants for the day and night start/end times somewhere so that the same times are used for anything else that needs that information. IIRC the only other use is time based evolutions.

Reply:
I'll make a PR for pret, as I feel that would be better for it.

Reply:
Actually, I would rather start defining those constants once we start looking into implementing DNS.